tests/charms/tls_certificates_interface/v1/test_tls_certificates_v1_requires.py

```diff
@@ -0,0 +1,594 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import json
+import unittest
+from datetime import datetime
+from unittest.mock import patch
+
+import pytest
+from ops import testing
+
+from tests.charms.tls_certificates_interface.v1.certificates import (
+    generate_ca as generate_ca_helper,
+)
+from tests.charms.tls_certificates_interface.v1.certificates import (
+    generate_certificate as generate_certificate_helper,
+)
+from tests.charms.tls_certificates_interface.v1.certificates import (
+    generate_csr as generate_csr_helper,
+)
+from tests.charms.tls_certificates_interface.v1.certificates import (
+    generate_private_key as generate_private_key_helper,
+)
+from tests.charms.tls_certificates_interface.v1.dummy_requirer_charm.src.charm import (
+    DummyTLSCertificatesRequirerCharm,
+)
+
+testing.SIMULATE_CAN_CONNECT = True
+
+BASE_CHARM_DIR = "tests.charms.tls_certificates_interface.v1.dummy_requirer_charm.src.charm.DummyTLSCertificatesRequirerCharm"  # noqa: E501
+LIB_DIR = "lib.charms.tls_certificates_interface.v1.tls_certificates"
+
+
+class Test(unittest.TestCase):
+    def setUp(self):
+        self.relation_name = "certificates"
+        self.harness = testing.Harness(DummyTLSCertificatesRequirerCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+        self.maxDiff = None
+
+    def test_given_no_relation_when_request_certificate_creation_then_runtime_error_is_raised(
+        self,
+    ):
+        with pytest.raises(RuntimeError):
+            self.harness.charm.certificates.request_certificate_creation(
+                certificate_signing_request=b"whatever csr"
+            )
+
+    def test_given_csr_when_request_certificate_creation_then_csr_is_sent_in_relation_data(self):
+        remote_app = "tls-certificates-provider"
+        relation_id = self.harness.add_relation(
```

Comment:
I created a method to cover relation creation. I haven't moved them to set up because I think the test author needs to be aware they are creating relations.